src/ngraph/op/variadic_split.cpp

```diff
@@ -0,0 +1,36 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/variadic_split.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+constexpr NodeTypeInfo op::v1::VariadicSplit::type_info;
+
+op::v1::VariadicSplit::VariadicSplit(const Output<Node>& data,
+                                     const Output<Node>& axis,
+                                     const Output<Node>& split_lengths)
+    : Op({data, axis, split_lengths})
+{
+    constructor_validate_and_infer_types();
```

Comment:
I see that validate_and_infer_types function is not implemented here, so how it will propagate output shapes and data type?

Reply:
I set all of the inputs to be relevant in output shape computation and set an output type if rank of `split_lengths` is known.
I can't calculate output shapes if `axis` and s`plit_lengths` are not constant.

Reply:
Ok, but if they are constants you can do it.